tests/utils.go

```diff
@@ -3816,12 +3838,48 @@ func RenderHostPathJob(jobName string, dir string, hostPathType k8sv1.HostPathTy
 	return job
 }
 
-// NewHelloWorldJob takes a DNS entry or an IP and a port which it will use create a pod
+func WaitForActualJobToSucceed(virtClient *kubecli.KubevirtClient, job *batchv1.Job, timeoutSec time.Duration) {
+	fail := false
+	EventuallyWithOffset(1, func() bool {
+		job, err := (*virtClient).BatchV1().Jobs(job.Namespace).Get(job.Name, metav1.GetOptions{})
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		if job.Status.Failed == *job.Spec.BackoffLimit {
+			fail = true
+			return fail
+		}
+		return job.Status.Succeeded > 0
+	}, timeoutSec*time.Second, 1*time.Second).Should(Equal(true))
```

Comment:
Could you please return the `job.Status` and then use `WithTransform` matcher to check the needed?
Then you can split the operation of fetching the data and asserting against the condition expected, plus providing the test more info if it fails (currently only a boolean will appear if this assertion fails).

Reply:
`WithTransform` takes the return value and passes it through a function that returns something you can assert against.
So you can return the status and then the transform can fetch you an inner value.
If it is not clear, I could try to convert it myself to this.

But maybe now that you changed the inner code with assertions it may no longer fit. I think you added an additional "early exit" assert.

Reply:
The logic before was actually "early exit". This is more oblivious I guess. I don't see a way to use `WithTransform` to "early exit"